group_project_v2/public/js/components/submission.js

```diff
@@ -3,12 +3,65 @@
 function GroupProjectSubmissionBlock(runtime, element) {
     "use strict";
 
-    function getMessageFromJson(jqXHR){
-        return jqXHR.responseJSON ? jqXHR.responseJSON.message : jqXHR.responseText;
-    }
+    /**
+     * This function is responsible for formatting the modal dialog for user.
+     */
+    function prepareMessageObject(jqXHR, default_title){
+
+        function getMessageFromJson(jqXHR){
+            return jqXHR.responseJSON ? jqXHR.responseJSON.message : jqXHR.responseText;
+        }
+
+        function getMessageTitleFromJson(jqXHR, default_title) {
+            return (jqXHR.responseJSON && jqXHR.responseJSON.title) ? jqXHR.responseJSON.title : default_title;
+        }
+
+        function filterMessageObject(message){
+            if(message.status === 0 && message.statusText === 'abort') {
+                message.title = GroupProjectCommon.gettext('Upload cancelled.');
+                message.content = GroupProjectCommon.gettext('Upload cancelled by user.');
+            }
+            if (message.status === 403) {
+                var base_message = GroupProjectCommon.gettext(
+                    "An error occurred while uploading your file. Please " +
+                    "refresh the page and try again. If it still does not " +
+                    "upload, please contact your Course TA."
+                );
+                var technical_details = '';
+                // Exact CSRF response message may vary, and may be different
+                // between different environments. We can assume that it should
+                // contain CSRF string.
+                if (message.content.indexOf('CSRF') !== -1){
+                    technical_details += GroupProjectCommon.gettext(' Technical details: CSRF verification failed.');
```

Comment:
Might it be possible to put this on its own line? Instead of:

```
An error occurred while uploading your file. Please refresh the page and try again. If it still does not upload, please contact your Course TA. Technical details: CSRF verification failed.
```
...do:

```
An error occurred while uploading your file. Please refresh the page and try again. If it still does not upload, please contact your Course TA.

Technical details: CSRF verification failed.
```

I don't know if you're able to safely inject HTML at this point or not or if it would break some sort of abstraction, but it would be much more presentable.